themes/bootstrap3/js/facets.js

```diff
@@ -116,18 +116,283 @@ VuFind.register('facetList', function FacetList() {
   return { setup: setup, getContent: getContent, updateContent: updateContent };
 });
 
+/* --- Multi Facets Handling --- */
+VuFind.register('multiFacetsSelection', function multiFacetsSelection() {
+  let globalAddedParams = [];
+  let globalRemovedParams = [];
+  let initialRawParams = window.location.search.substring(1).split('&');
+  let initialFilteredParams = initialRawParams.filter(function isFilter(obj) {
+    return obj.startsWith(encodeURI('filter[]='));
+  });
+  let dateSelectorId;
+  let isMultiFacetsSelectionActivated = false;
+  let callbackOnApply;
+  let callbackWhenDeactivated;
+  let defaultContext;
+
+  function stickApplyFiltersButtonAtTopWhenScrolling() {
+    let applyFilters = defaultContext.getElementsByClassName('apply-filters')[0];
+    let checkbox = defaultContext.getElementsByClassName('apply-filters-selection')[0];
+    window.onscroll = function fixButton() {
+      // To handle delayed loading elements changing the elements offset in the page
+      // We update the offset, depending on if we past the button or not
+      if (checkbox.getBoundingClientRect().bottom < 0) {
+        applyFilters.classList.add('fixed');
+      } else {
+        applyFilters.classList.remove('fixed');
+      }
+    };
+  }
+
+  function handleDateSelector() {
+    if (dateSelectorId === undefined) {
+      return;
+    }
+
+    let dateParams = [];
+    let allEmptyDateParams = true;
+    let form = document.querySelector('form#' + dateSelectorId);
+    let inputs = form.querySelectorAll('.date-fields input');
+    inputs.forEach(function checkDateParams(input) {
+      if (window.location.search.match(input.name)) {
+        // If the parameter is already present we update it
+        let count = initialRawParams.length;
+        for (let i = 0; i < count; i++) {
+          if (initialRawParams[i].startsWith(input.name + '=')) {
+            initialRawParams[i] = encodeURI(input.name + '=' + input.value); // Update
+            // If not empty we add it to date params
+            if (this.value !== '') {
+              allEmptyDateParams = false;
+            }
+            break;
+          }
+        }
+      } else {
+        dateParams.push(encodeURI(input.name + '=' + input.value));
+        if (this.value !== '') {
+          allEmptyDateParams = false;
+        }
+      }
+    });
+    // If at least one parameter is not null we continue the routine for the final URL
+    if (allEmptyDateParams === false) {
+      globalAddedParams = globalAddedParams.concat(dateParams);
+      let fieldName = form.querySelector('input[name="daterange[]"]').value;
```

Comment:
I did a bit of experimentation with non-default range configurations and ran into trouble here. If I turn on other types of range facets (see fullDateRange / numericRange / genericRange settings in facets.ini), I get a fatal error here in line 180, caused by the query selector returning null. I suspect that these special types of ranges will not be handled correctly in any case since there is no logic to handle the corresponding inputs.

If it would be helpful to add additional classes or other markup to all of the ranges so that they could be handled more simply/uniformly by this code, I would not object; otherwise, I think this could get very complicated and hard to maintain just to handle some rarely-used edge cases.

Let me know what you think!

Reply:
I made a modification to get the range name dynamically, let me know what you think of it

Reply:
That has eliminated the fatal error, but multiple range facets are still being ignored. Here's an example:

1.) In the [Results] section of facets.ini, add:

```
title = "Title Range"
topic = "Topic Range"
```

2.) In the SpecialFacets section of facets.ini, add:

```
genericRange[] = title
genericRange[] = topic
```

3.) Now, turn on multi-select, and put values in the two new range controls as well as the date control. When you submit, only one range value gets applied.

(I realize that this is a fairly rarely-used edge case, and the range facets I set up in my example are not at all useful -- I just created them to demonstrate the underlying functionality... if it's too complicated/difficult to make this fully functional, I would also be open to doing something else, like disabling range controls in multi-select mode).

Reply:
I changed it to have a dynamic number of daterange facets

Reply:
Thank you! This is looking better, but while testing, I discovered some bugs that I have addressed in #3919. I'll wait for that to be approved and merged before doing further testing here, just to be sure any issues I encounter are related to changes here and not pre-existing problems.